pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -454,7 +457,8 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(push *model.PushCont
 			buildSidecarOutboundListeners(configgen).
 			buildHTTPProxyListener(configgen).
 			buildVirtualOutboundListener(configgen).
-			buildVirtualInboundListener(configgen)
+			buildVirtualInboundListener(configgen).
+			buildSidecarDNSListener(configgen)
```

Comment:
@rshriram should we guard this with a feature flag instead of enabling it by default?

Reply:
Yes. this is WIP. Not done yet.

Reply:
Besides, this wont work on 1.6 proxies